server/controllers/user-auth.js

```diff
@@ -0,0 +1,33 @@
+
+import { hash } from 'bcrypt';
+import client from '../database/db_connect';
+
+export async function usersignup(req, resp) {
+    const {lastname , firstname , email , type} = req.body
+     let isadmin;
+     if (type == 'admin') {
+         isadmin = true
+     } else {
+         isadmin = false
+     }
+     try {
+         client.connect().then(console.log('database succesfully connected'))
+        const hashedpassword = await hash(req.body.password, 10)
+        client.query("INSERT INTO users(email , firstname , lastname , password , type , isadmin)  VALUES($1, $2 , $3 , $4 , $5 , $6)", [email, firstname, lastname, hashedpassword, type, isadmin], (err, rows) => {
+                 showdata(email);
```

Comment:
You don't need to query the database again! you can add `RETURNING *` to your query and the result will contain the created user